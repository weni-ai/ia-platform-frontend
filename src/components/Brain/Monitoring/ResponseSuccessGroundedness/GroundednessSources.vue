<template>
  <section
    v-for="fragment in fragments"
    :key="fragment.sentence"
    class="groundedness-sources"
  >
    <section class="groundedness-sources__source">
      <span
        :class="[
          'source__status',
          `source__status--${getReliabilityLevel(fragment)}`,
        ]"
      />
      <UnnnicIntelligenceText
        v-if="fragment.sources[0].filename"
        class="source__filename"
        color="neutral-dark"
        family="secondary"
        size="body-gt"
        tag="p"
      >
        {{ fragment.sources[0].filename }}
      </UnnnicIntelligenceText>
      <p
        v-else
        class="source__filename--not-found"
      >
        {{ $t('router.monitoring.inspect_response.not_found') }}
      </p>
      <UnnnicIntelligenceText
        class="source__confidence"
        color="neutral-cloudy"
        family="secondary"
        size="body-gt"
        tag="p"
      >
        {{
          $t('router.monitoring.inspect_response.confidence_percentage', {
            percentage: fragment.score,
          })
        }}
      </UnnnicIntelligenceText>
    </section>
  </section>
</template>

<script setup>
defineProps({
  fragments: {
    type: Array,
    required: true,
  },
  getReliabilityLevel: {
    type: Function,
    required: true,
  },
});
</script>

<style scoped lang="scss">
.groundedness-sources {
  display: grid;
  gap: $unnnic-spacing-xs;

  .groundedness-sources__source {
    display: grid;
    grid-template-columns: auto auto 1fr;
    gap: $unnnic-spacing-xs;
    align-items: center;

    .source__status {
      width: 10px;
      height: 10px;
      border-radius: 10px;
      background-color: $unnnic-color-neutral-cleanest;

      &--slightly-reliable {
        background-color: $unnnic-color-weni-300;
      }
      &--moderately-reliable {
        background-color: $unnnic-color-weni-600;
      }
      &--highly-reliable {
        background-color: $unnnic-color-weni-800;
      }
    }

    .source__filename {
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;

      &--not-found {
        color: $unnnic-color-neutral-dark;
        font-family: $unnnic-font-family-secondary;
        font-size: $unnnic-font-size-body-gt;
        line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
        font-style: italic;
      }
    }

    .source__confidence {
      white-space: nowrap;
    }
  }
}
</style>
