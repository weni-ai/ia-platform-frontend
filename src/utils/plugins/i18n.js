import * as VueI18n from 'vue-i18n';

import en from '../../locales/en.json';
import ptbr from '../../locales/pt_br.json';
import es from '../../locales/es.json';

const languages = {
  'en-US': en,
  'pt-BR': ptbr,
  es,
};

const numberFormats = {
  'en-US': {
    currency: {
      style: 'currency',
      currency: 'BRL',
      currencyDisplay: 'symbol',
    },
  },
  'pt-BR': {
    currency: {
      style: 'currency',
      currency: 'BRL',
      currencyDisplay: 'symbol',
    },
  },
  es: {
    currency: {
      style: 'currency',
      currency: 'BRL',
      currencyDisplay: 'symbol',
    },
  },
};

const messages = Object.assign(languages);

const i18n = VueI18n.createI18n({
  locale: navigator.language || navigator.userLanguage,
  fallbackLocale: 'en-US',
  messages,
  numberFormats,
});

export default i18n;
