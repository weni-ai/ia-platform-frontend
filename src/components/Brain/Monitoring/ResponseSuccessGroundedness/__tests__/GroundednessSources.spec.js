import { describe, it, expect, beforeEach, vi } from 'vitest';
import { shallowMount } from '@vue/test-utils';

import GroundednessSources from '../GroundednessSources.vue';
import UnnnicIntelligenceText from '@/components/unnnic-intelligence/Text.vue';
import i18n from '@/utils/plugins/i18n';

describe('GroundednessSources.vue', () => {
  let wrapper;

  const fragmentsMock = [
    {
      sentence: 'First sentence.',
      sources: [{ filename: 'source1.pdf' }],
      score: 85,
      reliability: 'high',
    },
    {
      sentence: 'Second sentence.',
      sources: [{ filename: '' }],
      score: 70,
      reliability: 'moderate',
    },
    {
      sentence: 'Third sentence.',
      sources: [{ filename: 'source3.docx' }],
      score: 95,
      reliability: 'very-high',
    },
  ];

  const getReliabilityLevelMock = vi.fn((fragment) => fragment.reliability);

  const mountWrapper = (props = {}) => {
    wrapper = shallowMount(GroundednessSources, {
      props: {
        fragments: fragmentsMock,
        getReliabilityLevel: getReliabilityLevelMock,
        ...props,
      },
      global: {
        stubs: {
          UnnnicIntelligenceText,
        },
      },
    });
  };

  beforeEach(() => {
    mountWrapper();
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders the correct number of sections for each fragment', () => {
    const sections = wrapper.findAll('[data-testid="source"]');
    expect(sections.length).toBe(fragmentsMock.length);
  });

  it('renders the reliability status class dynamically', () => {
    const sourcesStatus = wrapper.findAll('[data-testid="source-status"]');
    sourcesStatus.forEach((status, index) => {
      const reliabilityClass = `source__status--${fragmentsMock[index].reliability}`;
      expect(status.classes()).toContain(reliabilityClass);
    });
  });

  it('renders filename when available', () => {
    const filenames = wrapper.findAll('[data-testid="source-filename"]');
    expect(filenames.length).toBe(2);
    filenames.forEach((filename, index) => {
      expect(filename.text()).toBe(
        fragmentsMock[index === 1 ? 2 : 0].sources[0].filename,
      );
    });
  });

  it('renders "not found" text when filename is unavailable', () => {
    const notFoundTexts = wrapper.findAll(
      '[data-testid="source-filename-not-found"]',
    );
    expect(notFoundTexts.length).toBe(1);
    expect(notFoundTexts[0].text()).toBe(
      i18n.global.t('router.monitoring.inspect_response.not_found'),
    );
  });

  it('renders confidence percentage correctly', () => {
    const confidenceTexts = wrapper.findAll(
      '[data-testid="source-confidence"]',
    );
    confidenceTexts.forEach((confidence, index) => {
      expect(confidence.text()).toBe(
        i18n.global.t(
          'router.monitoring.inspect_response.confidence_percentage',
          {
            percentage: fragmentsMock[index].score,
          },
        ),
      );
    });
  });

  it('calls getReliabilityLevel for each fragment', () => {
    expect(getReliabilityLevelMock).toHaveBeenCalledTimes(fragmentsMock.length);
    fragmentsMock.forEach((fragment) => {
      expect(getReliabilityLevelMock).toHaveBeenCalledWith(fragment);
    });
  });

  it('renders no sections when fragments are empty', async () => {
    await wrapper.setProps({ fragments: [] });
    const sections = wrapper.findAll('data-testid="source"');
    expect(sections.length).toBe(0);
  });
});
