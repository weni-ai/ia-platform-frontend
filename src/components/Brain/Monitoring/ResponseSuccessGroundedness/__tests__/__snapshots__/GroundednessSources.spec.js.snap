// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GroundednessSources.vue > matches the snapshot 1`] = `
"<section data-v-4cf29c9a="" class="groundedness-sources" data-testid="source">
  <section data-v-4cf29c9a="" class="groundedness-sources__source"><span data-v-4cf29c9a="" data-testid="source-status" class="source__status source__status--high"></span>
    <p data-v-ec8da4ad="" data-v-4cf29c9a="" class="unnnic-text unnnic-text--size-body-gt unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-regular unnnic-text--color-neutral-dark source__filename source__filename" data-testid="source-filename">source1.pdf</p>
    <p data-v-ec8da4ad="" data-v-4cf29c9a="" class="unnnic-text unnnic-text--size-body-gt unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-regular unnnic-text--color-neutral-cloudy source__confidence source__confidence" data-testid="source-confidence">85% confidence</p>
  </section>
</section>
<section data-v-4cf29c9a="" class="groundedness-sources" data-testid="source">
  <section data-v-4cf29c9a="" class="groundedness-sources__source"><span data-v-4cf29c9a="" data-testid="source-status" class="source__status source__status--moderate"></span>
    <p data-v-4cf29c9a="" data-testid="source-filename-not-found" class="source__filename--not-found">Not found</p>
    <p data-v-ec8da4ad="" data-v-4cf29c9a="" class="unnnic-text unnnic-text--size-body-gt unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-regular unnnic-text--color-neutral-cloudy source__confidence source__confidence" data-testid="source-confidence">70% confidence</p>
  </section>
</section>
<section data-v-4cf29c9a="" class="groundedness-sources" data-testid="source">
  <section data-v-4cf29c9a="" class="groundedness-sources__source"><span data-v-4cf29c9a="" data-testid="source-status" class="source__status source__status--very-high"></span>
    <p data-v-ec8da4ad="" data-v-4cf29c9a="" class="unnnic-text unnnic-text--size-body-gt unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-regular unnnic-text--color-neutral-dark source__filename source__filename" data-testid="source-filename">source3.docx</p>
    <p data-v-ec8da4ad="" data-v-4cf29c9a="" class="unnnic-text unnnic-text--size-body-gt unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-regular unnnic-text--color-neutral-cloudy source__confidence source__confidence" data-testid="source-confidence">95% confidence</p>
  </section>
</section>"
`;
